### PR TITLE
Fix false positive for ``useless-super-delegation`` for variadics

### LINKED ISSUE #2270: False positive on useless-super-delegation with *args
### Steps to reproduce

Run pylint on the following code:
```python
class Super:
    def __init__(self, *args):
        self.args = args

class Sub(Super):
    def __init__(self, a, b):
        super().__init__(a, b)
```

### Current behavior

The following message is issued:
```
6:4: W0235: Useless super delegation in method '__init__' (useless-super-delegation)
```

### Expected behavior

I think no message should be issued, since the delegation is not useless: it forces the subclass to always take exactly 2 arguments, while the superclass can take any number of arguments.

This pattern occurs when defining a custom exception class, since the `Exception` class takes a variable number of arguments, but your custom exception type may have mandatory positional arguments.

### pylint --version output

```
pylint 2.0.0.dev2
astroid 2.0.0.dev4
Python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC]
```

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Write a good description on what the PR does.
- [x] Add an entry to the change log describing the change in
  `doc/whatsnew/2/2.15/index.rst` (or ``doc/whatsnew/2/2.14/full.rst``
   if the change needs backporting in 2.14). If necessary you can write
   details or offer examples on how the new change is supposed to work.
- [x] If you used multiple emails or multiple names when contributing, add your mails
      and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #2270.

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM !

File: tests/functional/u/useless/useless_super_delegation.py [hunk 1]
```diff
@@ -303,3 +303,13 @@ def __str__(self):
...
+
+
+# Reported in https://github.com/PyCQA/pylint/issues/2270
+class Super:
+    def __init__(self, *args):
+        self.args = args
+
+class Sub(Super):
+    def __init__(self, a, b):
+        super().__init__(a, b)
```
Comment (mbyrnepr2) [inline tests/functional/u/useless/useless_super_delegation.py:315]: How about when the first argument is required? I see the useless-super-delegation for this:

```python
class Super:
    def __init__(self, *args):
        self.args = args

class Sub(Super):
    def __init__(self, a, *args):
        super().__init__(a, *args)
```

Comment (DanielNoord) [reply tests/functional/u/useless/useless_super_delegation.py:315]: So you would want this to raise whenever there is only a variadic in `super()` and one or more positional in child?

Comment (mbyrnepr2) [reply tests/functional/u/useless/useless_super_delegation.py:315]: @DanielNoord that's exactly it. It's just a variation of the original issue which someone may encounter.

Comment (mbyrnepr2) [reply tests/functional/u/useless/useless_super_delegation.py:316]: I think that will work @DanielNoord. I haven't time and haven't played with other variations but I think it's good to go. Thanks!

File: pylint/checkers/classes/class_checker.py [hunk 2]
```diff
@@ -1276,6 +1276,12 @@ def _check_useless_super_delegation(self, function: nodes.FunctionDef) -> None:
         args = _signature_from_call(call)
 
         if meth_node is not None:
+            # Detect if the super method uses varargs and the function doesn't
```
Comment (DanielNoord) [inline pylint/checkers/classes/class_checker.py:1279]:
```suggestion
            # Detect if the super method uses varargs and the function doesn't or makes some of those explicit
```

File: tests/functional/u/useless/useless_super_delegation.py (hunk 1 above)
Comment (DanielNoord) [reply tests/functional/u/useless/useless_super_delegation.py:316]: Updated!

**Pull request merged**